Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8896,7 +8901,15 @@ void WebPageProxy::queryPermission(const ClientOrigin& clientOrigin, const Permi
         else if (*result == PermissionState::Prompt && shouldChangePromptToGrant)
             result = PermissionState::Granted;
         completionHandler(*result, false);
-    });
+    };
+
+    if (isOriginUnique(clientOrigin.topOrigin)) {
```

Comment:
Ok, I discussed it offline with Youenn and it sounds like we want to deal with these unique origins on the UIProcess side because we don't always return "Prompt" for these, unlike what the next line makes it look like. There is some logic above in this function that could cause us to return Granted or Denied for unique origins and we wouldn't want to duplicate that logic in the WebProcess.

I think we should add a isUnique() member function to SecurityOriginData (which makes sure that the members are empty and non-null) instead of this new isOriginUnique() free function. It might be useful in other places as it sounds like we sometimes want to determine in other processes than the WebProcess if an origin is unique (even though we cannot at the moment determine if 2 unique security origins are the same).

Reply:
Agreed, I think we should change isEmpty to isNull and add isUnique. Will do it in a follow-up.